### DISCUSSION TITLE
有办法取到package配置的默认值吗?

### DISCUSSION THREAD

**Discussion by DreamOfIce** [xmake-io/xmake#3650] [Q&A] [answered]
如题，目前在修改xrepo中的x264包，要实现仅当和默认值不一致才添加参数
有什么办法实现吗？

**Comment by waruqi** [comment]
if package:config("xxx") ~= "default val" then

**Reply by DreamOfIce** [reply]
量很大，我不希望每个都重复写一遍

**Reply by waruqi** [reply]
参考这个，走遍历 https://github.com/xmake-io/xmake-repo/blob/577278465e0d6f0683c3acd5e7e9f944bce33295/packages/f/ffmpeg/xmake.lua#L122

**Reply by DreamOfIce** [reply]
这个我知道，但是x264只接受和默认值不一样的参数，例如：
OpenCL 默认开启，可以传递 `--disable-opencl` 来关闭，但 `--enable-opencl ` 是非法的参数，会输出警告：
> Unknown option --enable-opencl, ignored

又如 lto 默认禁用，可以通过 `--enable-lto` 开启，但 `--disable-lto` 同样是非法的，也会输出警告

参数列表参见：
https://github.com/mirror/x264/blob/eaa68fad9e5d201d42fde51665f2d137ae96baf0/configure#L24-L58
参数处理部分位于：
https://github.com/mirror/x264/blob/eaa68fad9e5d201d42fde51665f2d137ae96baf0/configure#L418-L553

**Reply by waruqi** [reply, marked as answer]
遍历时候判断下 跟默认值相同，就不去设置无效参数。。如果不能完全通用，你只能挨个处理。。

另外，它只是警告后 ignore ，只要不影响编译，不影响配置值错误，有警告也无所谓，只是安装包，通常也就安装个一次，又不是经常本地编译调试。。

而且默认不开 -vD 也看不到警告

**Reply by DreamOfIce** [reply]
那么话说add_config可以在for循环里面动态添加嘛？
这样CLI可以正确显示选项吗？

**Comment by xq114** [comment]
```
local default_opts = {"opencl", ...}
for _, opt in ipairs(opts) do
	if table.contains(default_opts, opt) then ...
	else ...
	end
end
```
